market_api/market_api.py

```diff
@@ -29,11 +29,13 @@ def get_vin_history(self, vin):
             base_url=self.base_url,
             api_url=api_url
         )
+        req_query = self.get_querystring()
+        req_query['page'] = page
         response = requests.request("GET",
                                     request_url,
                                     headers=self.headers,
-                                    params=self.get_querystring())
-
+                                    params=req_query)
+        
         schema = GetVinHistoryOutputSchema()
         result = []
         for history in response.json():
```

Comment:
Does the response have the current_number of page and total number of count/page information in ?
If there is please add those in the response.  
That's why I was indicating to ask a pagination response class.

Reply:
At the time pagination response class not added... I have check marketapi docs and response not contain the total number of results & page number therefor the its not possible to guess the count pages information....

Reply:
pagination class can be apply if we can get all records once... 
What if i make for loop and make multiple request for marketsearch-api and fetch all data every time and then apply pagination class is it work for your.

In current i use keyword arg and add it in header of marketsearch-api request... That's according to marketsearch-api documentation.